frontend/src/pages/Preference Add/PrefrenceAdd.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Chip, Button } from "@mui/material";
+import AddIcon from "@mui/icons-material/Add";
+import {
+  addPreferenceSlicerFunc,
+  fetchCategoryList,
+} from "../../redux/user/userSlice";
+
+function PreferenceAdd() {
+  const dispatch = useDispatch();
+  const user = useSelector((state) => state.user);
+  const category = useSelector((state) => state.user.categoryList);
+  const [selectedCategories, setSelectedCategories] = useState([]);
+
+  const handleAddPreference = (categoryIds) => {
+    dispatch(
+      addPreferenceSlicerFunc({
+        token: user.data.token.access,
+        category: categoryIds,
+      })
+    );
+  };
+
+  const handleToggleCategory = (categoryId) => {
+    if (selectedCategories.includes(categoryId)) {
+      setSelectedCategories(
+        selectedCategories.filter((id) => id !== categoryId)
```

Comment:
use prevState instead of using state directly.

Reply:
done